### PR TITLE
111 us create and join rooms to send messages

### PR DESCRIPTION
**Pull request by cnstll**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [x] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

**Comment by cnstll** [comment]
Need reviews and testing before merging

**Comment by cnstll** [comment]
Write access could be done through websockets to update all connected user instantaneously.
Read access could be done through http requests

**Comment by cnstll** [comment]
Best trails for testing: 
- https://github.com/nestjs/nest/issues/1368
- https://github.com/nestjs/docs.nestjs.com/issues/97

File: backend/app/src/channel/channel.service.ts [hunk 1]
```diff
@@ -228,4 +233,198 @@ export class ChannelService {
...
+        },
+      });
+      return editedChannel;
+    } catch (error) {
+      return null;
+    }
+  }
+
+  async deleteChannelWS(userId: string, dto: DeleteChannelDto) {
+    //TODO Check role is ADMIN | OWNER
```
Comment (estelle-rcr) [inline backend/app/src/channel/channel.service.ts:392]: pour le delete channel, on va se reposer sur les 'leave channel' des users, si un user est le dernier a quitté un channel alors ca delete le channel en base

File: backend/app/src/channel/dto/createchannel.dto.ts [hunk 2]
```diff
@@ -2,6 +2,10 @@ import { ChannelType } from '@prisma/client';
 import { IsNotEmpty, IsOptional, IsString } from 'class-validator';
 
 export class CreateChannelDto {
+  @IsString()
+  @IsOptional()
```
Comment (estelle-rcr) [inline backend/app/src/channel/dto/createchannel.dto.ts:6]: est-ce que l'id en dto est lié à la creation que tu faisais ? car en ddb, il est obligatoire et sera créé dans tous les cas

File: backend/app/src/channel/dto/deleteChannel.dto.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { ChannelType } from '@prisma/client';
+import { IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+export class DeleteChannelDto {
+  @IsString()
+  @IsNotEmpty()
+  id: string;
+
```
Comment (estelle-rcr) [inline backend/app/src/channel/dto/deleteChannel.dto.ts:8]: pourquoi a-t-on besoin du type et du password pour le delete de channel ? le password ne va servir qu'au join channel

File: backend/app/src/channel/channel.service.ts [hunk 4]
```diff
@@ -228,4 +233,198 @@ export class ChannelService {
...
+    }
+  }
+
+  async joinChannelWS(
+    dto: JoinChannelDto,
+    userId: string,
+    clientSocket: Socket,
+  ) {
+    try {
+      /* Check the password is provided in the DTO for protected chan) */
```
Comment (estelle-rcr) [inline backend/app/src/channel/channel.service.ts:314]: On fera cette partie avec le hashing de password?

Comment (cnstll) [reply backend/app/src/channel/channel.service.ts:314]: yes !

File: backend/app/src/channel/channel.service.ts (hunk 1 above)
Comment (cnstll) [reply backend/app/src/channel/channel.service.ts:392]: je change la fonction pour une leaveChannel avec une vérification sur le nombre de user dans le channel

File: backend/app/src/channel/dto/createchannel.dto.ts (hunk 2 above)
Comment (cnstll) [reply backend/app/src/channel/dto/createchannel.dto.ts:6]: yes ! id supprimé du dto

**Review by estelle-rcr — APPROVED** [review]

**Pull request merged**